src/Agent.Listener/Configuration/AutoLogonConfigurationManager.cs

```diff
@@ -0,0 +1,283 @@
+#if OS_WINDOWS
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security.Principal;
+using System.Threading;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener.Configuration
+{
+    [ServiceLocator(Default = typeof(AutoLogonConfigurationManager))]
+    public interface IAutoLogonConfigurationManager : IAgentService
+    {
+        void Configure(CommandSettings command);
+        void UnConfigure();
+        bool RestartNeeded();
+        bool IsAutoLogonConfigured();
```

Comment:
This is getting consumed in the upgrade route as well. When we have to call upgrade, we will have to know if the agent has autologon configured so that we can change our actions accordingly.

---
In reply to: [118036075](https://github.com/Microsoft/vsts-agent/pull/985#discussion_r118036075) [](ancestors = 118036075)